src/main/ts/commands/Revive.ts

```diff
@@ -0,0 +1,70 @@
+import * as O from 'fp-ts/Option';
+import { pipe } from 'fp-ts/function';
+import { SimpleGit } from 'simple-git';
+import { ReviveArgs } from '../args/BeehiveArgs';
+import { createReleaseBranch, getBranchDetails, getReleaseBranchName, versionFromReleaseBranch } from '../core/BranchLogic';
+import * as AdvanceVersion from '../core/AdvanceVersion';
+import { printHeaderMessage } from '../core/Messages';
+import * as Version from '../core/Version';
+import * as ArrayUtils from '../utils/ArrayUtils';
+import { Comparison } from '../utils/Comparison';
+import * as OptionUtils from '../utils/OptionUtils';
+import * as PromiseUtils from '../utils/PromiseUtils';
+import * as Git from '../utils/Git';
+
+type MajorMinorVersion = Version.MajorMinorVersion;
+type Version = Version.Version;
+
+interface TagDetails {
+  readonly version: Version;
+  readonly name: string;
+}
+
+const getLatestTag = async (git: SimpleGit, version: MajorMinorVersion): Promise<TagDetails> => {
+  const tags = await Git.getTags(git);
+  const tagDetails = OptionUtils.somes(tags.map((tag) => pipe(
+    Version.parseVersionE(tag),
+    O.fromEither,
+    O.filter((v) => Version.compareMajorMinorVersions(v, version) === Comparison.EQ),
+    O.map((v) => ({
+      version: v,
+      name: tag
+    }))
+  )));
```

Comment:
Earlier I said that you could implement `OptionUtils.somes` in terms of `Array.filterMap` — you could probably instead just directly use `Array.filterMap` if there was going to be some degree of mapping at the call site anyway.

That said, the curried nature of the fp-ts function means it might play better with typescript if you made a wrapper around it.